.github/workflows/ci.yml

```diff
@@ -43,3 +43,14 @@ jobs:
           otp-version: ${{matrix.otp}}
           rebar3-version: ${{matrix.rebar3}}
       - run: rebar3 do eunit, ct --cover, cover
+
+  dialyzer:
+    name: Run Dialyzer
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: "latest"
+          rebar3-version: "latest"
+      - run: rebar3 dialyzer
```

Comment:
It seems like this ran pretty quickly, but we may want to cache the plts in the future, although since Luerl doesn't have any dependencies I'm not sure if relevant